### PR TITLE
allow impersonation of users in the cli

### PR DESCRIPTION
**Pull request by matko**
The CLI does all its actions as admin. It's convenient to be able to perform them as another user too, so that authorization checks run. This PR introduces an `--impersonate` flag which can be used to pass in the user that is performing the action.

### CODE REVIEW

File: docs/terminusdb.1.ronn [hunk 1]
```diff
@@ -27,838 +27,14 @@ Display help regarding terminusdb.
 
 ### test
 
-`terminusdb test OPTIONS`
```
Comment (rrooij) [inline docs/terminusdb.1.ronn:30]: What happened to all the remove docs in the manpages?

**Review by rrooij — DISMISSED** [review]
Something weird went on with the manpage

File: docs/terminusdb.1.ronn (hunk 1 above)
Comment (matko) [reply docs/terminusdb.1.ronn:30]: oh that's interesting. This seems to be an issue with the automatic manpage generation. I'm not sure what's going on there.

Comment (matko) [reply docs/terminusdb.1.ronn:30]: It's all coming from this automated commit: https://github.com/terminusdb/terminusdb/pull/1709/commits/a11d4aa7f54be8abf90412b439cdf4011d3ec182

Comment (rrooij) [reply docs/terminusdb.1.ronn:30]: I see why it doesn't work. Your changes introduced a regression in markdown generation of the help page.

Try `./terminusdb help --markdown` in your branch vs `main`.

Comment (matko) [reply docs/terminusdb.1.ronn:30]: Looks like the markdown generation wasn't able to handle a flag that didn't have a short version.
I hope I just fixed that.

Comment (matko) [reply docs/terminusdb.1.ronn]: yup, manpage is back.

**Pull request merged**